SQL import - commented out lines not ignored

**Describe the bug**
I use a sql file to load my test data into my database(H2). For test purposes I comment out some datasets but then Hibernate cannot load the sql-file correctly. There are linebreaks in my queries to make them readable, so they are not in one line of text. for this reason I need to comment out multiple rows.

**Expected behavior**
Any line starting with the comment string "--" should be ignored.

**Actual behavior**
An exception was thrown while executing the queries.
```
javax.persistence.PersistenceException: [PersistenceUnit: default] Unable to build Hibernate SessionFactory
	at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.persistenceException(FastBootEntityManagerFactoryBuilder.java:113)
	at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.build(FastBootEntityManagerFactoryBuilder.java:67)
	at io.quarkus.hibernate.orm.runtime.FastBootHibernatePersistenceProvider.createEntityManagerFactory(FastBootHibernatePersistenceProvider.java:54)
	at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:80)
	at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:55)
	at io.quarkus.hibernate.orm.runtime.JPAConfig$LazyPersistenceUnit.get(JPAConfig.java:110)
	at io.quarkus.hibernate.orm.runtime.JPAConfig.startAll(JPAConfig.java:58)
	at io.quarkus.hibernate.orm.runtime.HibernateOrmRecorder.startAllPersistenceUnits(HibernateOrmRecorder.java:82)
	at io.quarkus.deployment.steps.HibernateOrmProcessor$startPersistenceUnits28.deploy_0(HibernateOrmProcessor$startPersistenceUnits28.zig:70)
	at io.quarkus.deployment.steps.HibernateOrmProcessor$startPersistenceUnits28.deploy(HibernateOrmProcessor$startPersistenceUnits28.zig:36)
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:242)
	at io.quarkus.runtime.Application.start(Application.java:87)
	at io.quarkus.runner.RuntimeRunner.run(RuntimeRunner.java:167)
	at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:249)
	at io.quarkus.test.junit.QuarkusTestExtension.createTestInstance(QuarkusTestExtension.java:394)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeTestInstanceFactory(ClassBasedTestDescriptor.java:285)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.instantiateTestClass(ClassBasedTestDescriptor.java:275)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.instantiateTestClass(ClassTestDescriptor.java:77)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.instantiateAndPostProcessTestInstance(ClassBasedTestDescriptor.java:258)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$testInstancesProvider$2(ClassBasedTestDescriptor.java:252)
	at java.base/java.util.Optional.orElseGet(Optional.java:369)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$testInstancesProvider$3(ClassBasedTestDescriptor.java:251)
	at org.junit.jupiter.engine.execution.TestInstancesProvider.getTestInstances(TestInstancesProvider.java:29)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$prepare$0(TestMethodTestDescriptor.java:106)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.prepare(TestMethodTestDescriptor.java:105)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.prepare(TestMethodTestDescriptor.java:69)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$prepare$1(NodeTestTask.java:107)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.prepare(NodeTestTask.java:107)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:75)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:229)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:197)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:211)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:191)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:230)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:58)
Caused by: org.hibernate.tool.hbm2ddl.ImportScriptException: Error during import script parsing.
	at org.hibernate.tool.hbm2ddl.MultipleLinesSqlCommandExtractor.extractCommands(MultipleLinesSqlCommandExtractor.java:31)
	at org.hibernate.tool.schema.internal.exec.AbstractScriptSourceInput.read(AbstractScriptSourceInput.java:40)
	at org.hibernate.tool.schema.internal.SchemaCreatorImpl.applyImportSources(SchemaCreatorImpl.java:491)
	at org.hibernate.tool.schema.internal.SchemaCreatorImpl.performCreation(SchemaCreatorImpl.java:180)
	at org.hibernate.tool.schema.internal.SchemaCreatorImpl.doCreation(SchemaCreatorImpl.java:135)
	at org.hibernate.tool.schema.internal.SchemaCreatorImpl.doCreation(SchemaCreatorImpl.java:121)
	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.performDatabaseAction(SchemaManagementToolCoordinator.java:156)
	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.process(SchemaManagementToolCoordinator.java:73)
	at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:320)
	at org.hibernate.boot.internal.SessionFactoryBuilderImpl.build(SessionFactoryBuilderImpl.java:462)
	at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.build(FastBootEntityManagerFactoryBuilder.java:65)
	... 61 more
Caused by: org.hibernate.hql.internal.antlr.SqlStatementParser$StatementParserException: Import script Sql statements must terminate with a ';' char
	at org.hibernate.hql.internal.antlr.SqlStatementParser.throwExceptionIfErrorOccurred(SqlStatementParser.java:55)
	at org.hibernate.tool.hbm2ddl.MultipleLinesSqlCommandExtractor.extractCommands(MultipleLinesSqlCommandExtractor.java:28)
	... 71 more
```


**To Reproduce**
Steps to reproduce the behavior:
1.  create an entity
2.  create an sql-script with some insert statements with line breaks
3.  comment out one multi line dataset
4.  add config to let quarkus load the sql
5.  start dev mode or a test to load the file
```
INSERT INTO public.youTable(...) 
    VALUES (...);
--INSERT INTO public.youTable(...) 
--    VALUES (...);
```


**Configuration**
```properties
quarkus.http.cors=true
quarkus.http.cors.origins=http://localhost:8080,http://localhost:4200

mp.openapi.scan.exclude.packages=java.util

quarkus.datasource.url=jdbc:h2:mem:test
quarkus.datasource.driver=org.h2.Driver
quarkus.datasource.min-size=1
quarkus.datasource.max-size=4

%dev.quarkus.hibernate-orm.database.generation = drop-and-create
%test.quarkus.hibernate-orm.database.generation = drop-and-create
%dev.quarkus.hibernate-orm.sql-load-script = import-dev.sql
%test.quarkus.hibernate-orm.sql-load-script = import-dev.sql
%dev.quarkus.hibernate-orm.log.sql=true
%test.quarkus.hibernate-orm.log.sql=true
```


**Environment (please complete the following information):**
 - Output of `java -version`: 
```
java 11.0.1 2018-10-16 LTS
Java(TM) SE Runtime Environment 18.9 (build 11.0.1+13-LTS)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.1+13-LTS, mixed mode)
```
 - GraalVM version (if different from Java): no GraalVM
 - Quarkus version or git rev: 1.2.1.Final
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 
```
Apache Maven 3.6.1 (d66c9c0b3152b2e69ee9bac180bb8fcc8e6af555; 2019-04-04T21:00:29+02:00)
Maven home: C:\Users\...\.m2\wrapper\dists\apache-maven-3.6.1-bin\38pn40mp89t5c94bjdbeod370m\apache-maven-3.6.1
Java version: 11.0.1, vendor: Oracle Corporation, runtime: C:\Program Files\Java\jdk-11.0.1
Default locale: de_DE, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"
```

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -84,7 +84,7 @@
         <commons-lang3.version>3.9</commons-lang3.version>
         <commons-codec.version>1.14</commons-codec.version>
         <classmate.version>1.3.4</classmate.version>
-        <hibernate-orm.version>5.4.19.Final</hibernate-orm.version>
+        <hibernate-orm.version>5.4.21.Final</hibernate-orm.version>
         <hibernate-reactive.version>1.0.0.Alpha8</hibernate-reactive.version>
         <hibernate-validator.version>6.1.5.Final</hibernate-validator.version>
         <hibernate-search.version>6.0.0.Beta9</hibernate-search.version>

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -85,7 +85,7 @@
         <commons-codec.version>1.14</commons-codec.version>
         <classmate.version>1.3.4</classmate.version>
         <hibernate-orm.version>5.4.21.Final</hibernate-orm.version>
-        <hibernate-reactive.version>1.0.0.Alpha8</hibernate-reactive.version>
+        <hibernate-reactive.version>1.0.0.Alpha9</hibernate-reactive.version>
         <hibernate-validator.version>6.1.5.Final</hibernate-validator.version>
         <hibernate-search.version>6.0.0.Beta9</hibernate-search.version>
         <narayana.version>5.10.5.Final</narayana.version>

--- a/extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/ReactiveSessionProducer.java
+++ b/extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/ReactiveSessionProducer.java
@@ -1,7 +1,5 @@
 package io.quarkus.hibernate.reactive.runtime;
 
-import java.util.concurrent.CompletionStage;
-
 import javax.enterprise.context.ApplicationScoped;
 import javax.enterprise.context.RequestScoped;
 import javax.enterprise.inject.Disposes;
@@ -12,7 +10,6 @@ import org.hibernate.reactive.mutiny.Mutiny;
 import org.hibernate.reactive.stage.Stage;
 
 import io.quarkus.arc.DefaultBean;
-import io.smallrye.mutiny.Uni;
 
 @ApplicationScoped
 public class ReactiveSessionProducer {
@@ -26,40 +23,15 @@ public class ReactiveSessionProducer {
     @Produces
     @RequestScoped
     @DefaultBean
-    public CompletionStage<Stage.Session> stageSession() {
+    public Stage.Session createStageSession() {
         return reactiveSessionFactory.openSession();
     }
 
     @Produces
     @RequestScoped
     @DefaultBean
-    public Uni<Mutiny.Session> mutinySession() {
-        return mutinySessionFactory.openSession().cache();
-    }
-
-    public void disposeStageSession(@Disposes CompletionStage<Stage.Session> reactiveSession) {
-        reactiveSession.whenComplete((s, t) -> {
-            if (s != null)
-                s.close();
-        });
-    }
-
-    public void disposeMutinySession(@Disposes Uni<Mutiny.Session> reactiveSession) {
-        reactiveSession.subscribe().with(Mutiny.Session::close);
-    }
-
-    @Produces
-    @RequestScoped
-    @DefaultBean
-    public Stage.Session createStageSession() {
-        return reactiveSessionFactory.createSession();
-    }
-
-    @Produces
-    @RequestScoped
-    @DefaultBean
     public Mutiny.Session createMutinySession() {
-        return mutinySessionFactory.createSession();
+        return mutinySessionFactory.openSession();
     }
 
     public void disposeStageSession(@Disposes Stage.Session reactiveSession) {

--- a/integration-tests/hibernate-reactive-db2/src/main/java/io/quarkus/it/hibernate/reactive/db2/HibernateReactiveDB2TestEndpoint.java
+++ b/integration-tests/hibernate-reactive-db2/src/main/java/io/quarkus/it/hibernate/reactive/db2/HibernateReactiveDB2TestEndpoint.java
@@ -21,10 +21,10 @@ import io.vertx.mutiny.sqlclient.Tuple;
 public class HibernateReactiveDB2TestEndpoint {
 
     @Inject
-    CompletionStage<Stage.Session> stageSession;
+    Stage.Session stageSession;
 
     @Inject
-    Uni<Mutiny.Session> mutinySession;
+    Mutiny.Session mutinySession;
 
     // Injecting a Vert.x Pool is not required, it us only used to
     // independently validate the contents of the database for the test
@@ -36,11 +36,8 @@ public class HibernateReactiveDB2TestEndpoint {
     @Produces(MediaType.APPLICATION_JSON)
     public CompletionStage<GuineaPig> reactiveFind() {
         final GuineaPig expectedPig = new GuineaPig(5, "Aloi");
-        return stageSession
-                .thenCompose(session -> {
-                    return populateDB().convert().toCompletionStage()
-                            .thenCompose(junk -> session.find(GuineaPig.class, expectedPig.getId()));
-                });
+        return populateDB().convert().toCompletionStage()
+                .thenCompose(junk -> stageSession.find(GuineaPig.class, expectedPig.getId()));
     }
 
     @GET
@@ -48,20 +45,16 @@ public class HibernateReactiveDB2TestEndpoint {
     @Produces(MediaType.APPLICATION_JSON)
     public Uni<GuineaPig> reactiveFindMutiny() {
         final GuineaPig expectedPig = new GuineaPig(5, "Aloi");
-        return mutinySession
-                .flatMap(session -> {
-                    return populateDB()
-                            .then(() -> session.find(GuineaPig.class, expectedPig.getId()));
-                });
+        return populateDB()
+                .then(() -> mutinySession.find(GuineaPig.class, expectedPig.getId()));
     }
 
     @GET
     @Path("/reactivePersist")
     @Produces(MediaType.APPLICATION_JSON)
     public Uni<String> reactivePersist() {
-        return mutinySession
-                .flatMap(s -> s.persist(new GuineaPig(10, "Tulip")))
-                .flatMap(s -> s.flush())
+        return mutinySession.persist(new GuineaPig(10, "Tulip"))
+                .chain(s -> s.flush())
                 .then(() -> selectNameFromId(10));
     }
 
@@ -69,46 +62,33 @@ public class HibernateReactiveDB2TestEndpoint {
     @Path("/reactiveRemoveTransientEntity")
     @Produces(MediaType.APPLICATION_JSON)
     public Uni<String> reactiveRemoveTransientEntity() {
-        return mutinySession
-                .flatMap(mutinySession -> {
-                    return populateDB()
-                            .flatMap(junk -> selectNameFromId(5))
-                            .map(name -> {
-                                if (name == null)
-                                    throw new AssertionError("Database was not populated properly");
-                                return name;
-                            })
-                            .flatMap(junk -> mutinySession.merge(new GuineaPig(5, "Aloi")))
-                            .flatMap(aloi -> mutinySession.remove(aloi))
-                            .flatMap(junk -> mutinySession.flush())
-                            .flatMap(junk -> selectNameFromId(5))
-                            .map(result -> {
-                                if (result == null)
-                                    return "OK";
-                                else
-                                    return result;
-                            });
-                });
+        return populateDB()
+                .then(() -> selectNameFromId(5))
+                .map(name -> {
+                    if (name == null) {
+                        throw new AssertionError("Database was not populated properly");
+                    }
+                    return name;
+                })
+                .then(() -> mutinySession.merge(new GuineaPig(5, "Aloi")))
+                .chain(aloi -> mutinySession.remove(aloi))
+                .then(() -> mutinySession.flush())
+                .then(() -> selectNameFromId(5))
+                .onItem().ifNotNull().transform(result -> result)
+                .onItem().ifNull().continueWith("OK");
     }
 
     @GET
     @Path("/reactiveRemoveManagedEntity")
     @Produces(MediaType.APPLICATION_JSON)
     public Uni<String> reactiveRemoveManagedEntity() {
-        return mutinySession
-                .flatMap(mutinySession -> {
-                    return populateDB()
-                            .flatMap(junk -> mutinySession.find(GuineaPig.class, 5))
-                            .flatMap(aloi -> mutinySession.remove(aloi))
-                            .flatMap(junk -> mutinySession.flush())
-                            .flatMap(junk -> selectNameFromId(5))
-                            .map(result -> {
-                                if (result == null)
-                                    return "OK";
-                                else
-                                    return result;
-                            });
-                });
+        return populateDB()
+                .then(() -> mutinySession.find(GuineaPig.class, 5))
+                .chain(aloi -> mutinySession.remove(aloi))
+                .then(() -> mutinySession.flush())
+                .then(() -> selectNameFromId(5))
+                .onItem().ifNotNull().transform(result -> result)
+                .onItem().ifNull().continueWith("OK");
     }
 
     @GET
@@ -116,24 +96,22 @@ public class HibernateReactiveDB2TestEndpoint {
     @Produces(MediaType.APPLICATION_JSON)
     public Uni<String> reactiveUpdate() {
         final String NEW_NAME = "Tina";
-        return mutinySession
-                .flatMap(mutinySession -> {
-                    return populateDB()
-                            .flatMap(junk -> mutinySession.find(GuineaPig.class, 5))
-                            .map(pig -> {
-                                if (NEW_NAME.equals(pig.getName()))
-                                    throw new AssertionError("Pig already had name " + NEW_NAME);
-                                pig.setName(NEW_NAME);
-                                return pig;
-                            })
-                            .flatMap(junk -> mutinySession.flush())
-                            .flatMap(junk -> selectNameFromId(5));
-                });
+        return populateDB()
+                .then(() -> mutinySession.find(GuineaPig.class, 5))
+                .map(pig -> {
+                    if (NEW_NAME.equals(pig.getName())) {
+                        throw new AssertionError("Pig already had name " + NEW_NAME);
+                    }
+                    pig.setName(NEW_NAME);
+                    return pig;
+                })
+                .then(() -> mutinySession.flush())
+                .then(() -> selectNameFromId(5));
     }
 
     private Uni<RowSet<Row>> populateDB() {
         return db2Pool.query("DELETE FROM Pig").execute()
-                .flatMap(junk -> db2Pool.preparedQuery("INSERT INTO Pig (id, name) VALUES (5, 'Aloi')").execute());
+                .then(() -> db2Pool.preparedQuery("INSERT INTO Pig (id, name) VALUES (5, 'Aloi')").execute());
     }
 
     private Uni<String> selectNameFromId(Integer id) {

--- a/integration-tests/hibernate-reactive-mysql/src/main/java/io/quarkus/it/hibernate/reactive/mysql/HibernateReactiveMySQLTestEndpoint.java
+++ b/integration-tests/hibernate-reactive-mysql/src/main/java/io/quarkus/it/hibernate/reactive/mysql/HibernateReactiveMySQLTestEndpoint.java
@@ -21,10 +21,10 @@ import io.vertx.mutiny.sqlclient.Tuple;
 public class HibernateReactiveMySQLTestEndpoint {
 
     @Inject
-    CompletionStage<Stage.Session> stageSession;
+    Stage.Session stageSession;
 
     @Inject
-    Uni<Mutiny.Session> mutinySession;
+    Mutiny.Session mutinySession;
 
     // Injecting a Vert.x Pool is not required, it us only used to
     // independently validate the contents of the database for the test
@@ -36,11 +36,8 @@ public class HibernateReactiveMySQLTestEndpoint {
     @Produces(MediaType.APPLICATION_JSON)
     public CompletionStage<GuineaPig> reactiveFind() {
         final GuineaPig expectedPig = new GuineaPig(5, "Aloi");
-        return stageSession
-                .thenCompose(session -> {
-                    return populateDB().convert().toCompletionStage()
-                            .thenCompose(junk -> session.find(GuineaPig.class, expectedPig.getId()));
-                });
+        return populateDB().convert().toCompletionStage()
+                .thenCompose(junk -> stageSession.find(GuineaPig.class, expectedPig.getId()));
     }
 
     @GET
@@ -48,67 +45,50 @@ public class HibernateReactiveMySQLTestEndpoint {
     @Produces(MediaType.APPLICATION_JSON)
     public Uni<GuineaPig> reactiveFindMutiny() {
         final GuineaPig expectedPig = new GuineaPig(5, "Aloi");
-        return mutinySession
-                .flatMap(session -> {
-                    return populateDB()
-                            .then(() -> session.find(GuineaPig.class, expectedPig.getId()));
-                });
+        return populateDB()
+                .then(() -> mutinySession.find(GuineaPig.class, expectedPig.getId()));
     }
 
     @GET
     @Path("/reactivePersist")
     @Produces(MediaType.APPLICATION_JSON)
     public Uni<String> reactivePersist() {
-        return mutinySession
-                .flatMap(s -> s.persist(new GuineaPig(10, "Tulip")))
-                .flatMap(s -> s.flush())
-                .flatMap(junk -> selectNameFromId(10));
+        return mutinySession.persist(new GuineaPig(10, "Tulip"))
+                .chain(s -> s.flush())
+                .then(() -> selectNameFromId(10));
     }
 
     @GET
     @Path("/reactiveRemoveTransientEntity")
     @Produces(MediaType.APPLICATION_JSON)
     public Uni<String> reactiveRemoveTransientEntity() {
-        return mutinySession
-                .flatMap(mutinySession -> {
-                    return populateDB()
-                            .flatMap(junk -> selectNameFromId(5))
-                            .map(name -> {
-                                if (name == null)
-                                    throw new AssertionError("Database was not populated properly");
-                                return name;
-                            })
-                            .flatMap(junk -> mutinySession.merge(new GuineaPig(5, "Aloi")))
-                            .flatMap(aloi -> mutinySession.remove(aloi))
-                            .flatMap(junk -> mutinySession.flush())
-                            .flatMap(junk -> selectNameFromId(5))
-                            .map(result -> {
-                                if (result == null)
-                                    return "OK";
-                                else
-                                    return result;
-                            });
-                });
+        return populateDB()
+                .then(() -> selectNameFromId(5))
+                .map(name -> {
+                    if (name == null) {
+                        throw new AssertionError("Database was not populated properly");
+                    }
+                    return name;
+                })
+                .then(() -> mutinySession.merge(new GuineaPig(5, "Aloi")))
+                .chain(aloi -> mutinySession.remove(aloi))
+                .then(() -> mutinySession.flush())
+                .then(() -> selectNameFromId(5))
+                .onItem().ifNotNull().transform(result -> result)
+                .onItem().ifNull().continueWith("OK");
     }
 
     @GET
     @Path("/reactiveRemoveManagedEntity")
     @Produces(MediaType.APPLICATION_JSON)
     public Uni<String> reactiveRemoveManagedEntity() {
-        return mutinySession
-                .flatMap(mutinySession -> {
-                    return populateDB()
-                            .flatMap(junk -> mutinySession.find(GuineaPig.class, 5))
-                            .flatMap(aloi -> mutinySession.remove(aloi))
-                            .flatMap(junk -> mutinySession.flush())
-                            .flatMap(junk -> selectNameFromId(5))
-                            .map(result -> {
-                                if (result == null)
-                                    return "OK";
-                                else
-                                    return result;
-                            });
-                });
+        return populateDB()
+                .then(() -> mutinySession.find(GuineaPig.class, 5))
+                .chain(aloi -> mutinySession.remove(aloi))
+                .then(() -> mutinySession.flush())
+                .then(() -> selectNameFromId(5))
+                .onItem().ifNotNull().transform(result -> result)
+                .onItem().ifNull().continueWith("OK");
     }
 
     @GET
@@ -116,19 +96,17 @@ public class HibernateReactiveMySQLTestEndpoint {
     @Produces(MediaType.APPLICATION_JSON)
     public Uni<String> reactiveUpdate() {
         final String NEW_NAME = "Tina";
-        return mutinySession
-                .flatMap(mutinySession -> {
-                    return populateDB()
-                            .flatMap(junk -> mutinySession.find(GuineaPig.class, 5))
-                            .map(pig -> {
-                                if (NEW_NAME.equals(pig.getName()))
-                                    throw new AssertionError("Pig already had name " + NEW_NAME);
-                                pig.setName(NEW_NAME);
-                                return pig;
-                            })
-                            .flatMap(junk -> mutinySession.flush())
-                            .flatMap(junk -> selectNameFromId(5));
-                });
+        return populateDB()
+                .then(() -> mutinySession.find(GuineaPig.class, 5))
+                .map(pig -> {
+                    if (NEW_NAME.equals(pig.getName())) {
+                        throw new AssertionError("Pig already had name " + NEW_NAME);
+                    }
+                    pig.setName(NEW_NAME);
+                    return pig;
+                })
+                .then(() -> mutinySession.flush())
+                .then(() -> selectNameFromId(5));
     }
 
     private Uni<RowSet<Row>> populateDB() {

--- a/integration-tests/hibernate-reactive-postgresql/src/main/java/io/quarkus/it/hibernate/reactive/postgresql/HibernateReactiveTestEndpoint.java
+++ b/integration-tests/hibernate-reactive-postgresql/src/main/java/io/quarkus/it/hibernate/reactive/postgresql/HibernateReactiveTestEndpoint.java
@@ -21,10 +21,10 @@ import io.vertx.mutiny.sqlclient.Tuple;
 public class HibernateReactiveTestEndpoint {
 
     @Inject
-    CompletionStage<Stage.Session> stageSession;
+    Stage.Session stageSession;
 
     @Inject
-    Uni<Mutiny.Session> mutinySession;
+    Mutiny.Session mutinySession;
 
     // Injecting a Vert.x Pool is not required, it us only used to
     // independently validate the contents of the database for the test
@@ -36,11 +36,8 @@ public class HibernateReactiveTestEndpoint {
     @Produces(MediaType.APPLICATION_JSON)
     public CompletionStage<GuineaPig> reactiveFind() {
         final GuineaPig expectedPig = new GuineaPig(5, "Aloi");
-        return stageSession
-                .thenCompose(session -> {
-                    return populateDB().convert().toCompletionStage()
-                            .thenCompose(junk -> session.find(GuineaPig.class, expectedPig.getId()));
-                });
+        return populateDB().convert().toCompletionStage()
+                .thenCompose(junk -> stageSession.find(GuineaPig.class, expectedPig.getId()));
     }
 
     @GET
@@ -48,21 +45,18 @@ public class HibernateReactiveTestEndpoint {
     @Produces(MediaType.APPLICATION_JSON)
     public Uni<GuineaPig> reactiveFindMutiny() {
         final GuineaPig expectedPig = new GuineaPig(5, "Aloi");
-        return mutinySession
-                .flatMap(session -> {
-                    return populateDB()
-                            .then(() -> session.find(GuineaPig.class, expectedPig.getId()));
-                });
+        return populateDB().then(() -> mutinySession.find(GuineaPig.class, expectedPig.getId()));
     }
 
     @GET
     @Path("/reactivePersist")
     @Produces(MediaType.APPLICATION_JSON)
     public Uni<String> reactivePersist() {
+        final GuineaPig pig = new GuineaPig(10, "Tulip");
         return mutinySession
-                .flatMap(s -> s.persist(new GuineaPig(10, "Tulip")))
-                .flatMap(s -> s.flush())
-                .flatMap(junk -> selectNameFromId(10));
+                .persist(pig)
+                .then(() -> mutinySession.flush())
+                .then(() -> selectNameFromId(10));
     }
 
     @GET
@@ -72,8 +66,8 @@ public class HibernateReactiveTestEndpoint {
         final FriesianCow cow = new FriesianCow();
         cow.name = "Carolina";
         return mutinySession
-                .flatMap(s -> s.persist(cow))
-                .flatMap(s -> s.flush())
+                .persist(cow)
+                .then(() -> mutinySession.flush())
                 .flatMap(s -> s.createQuery("from FriesianCow f where f.name = :name", FriesianCow.class)
                         .setParameter("name", cow.name).getSingleResult());
     }
@@ -82,25 +76,21 @@ public class HibernateReactiveTestEndpoint {
     @Path("/reactiveRemoveTransientEntity")
     @Produces(MediaType.APPLICATION_JSON)
     public Uni<String> reactiveRemoveTransientEntity() {
-        return mutinySession
-                .flatMap(mutinySession -> {
-                    return populateDB()
-                            .flatMap(junk -> selectNameFromId(5))
-                            .map(name -> {
-                                if (name == null)
-                                    throw new AssertionError("Database was not populated properly");
-                                return name;
-                            })
-                            .flatMap(junk -> mutinySession.merge(new GuineaPig(5, "Aloi")))
-                            .flatMap(aloi -> mutinySession.remove(aloi))
-                            .flatMap(junk -> mutinySession.flush())
-                            .flatMap(junk -> selectNameFromId(5))
-                            .map(result -> {
-                                if (result == null)
-                                    return "OK";
-                                else
-                                    return result;
-                            });
+        return populateDB()
+                .then(() -> selectNameFromId(5))
+                .invoke(name -> {
+                    if (name == null)
+                        throw new AssertionError("Database was not populated properly");
+                })
+                .then(() -> mutinySession.merge(new GuineaPig(5, "Aloi")))
+                .invoke(aloi -> mutinySession.remove(aloi))
+                .then(() -> mutinySession.flush())
+                .then(() -> selectNameFromId(5))
+                .map(result -> {
+                    if (result == null)
+                        return "OK";
+                    else
+                        return result;
                 });
     }
 
@@ -108,19 +98,16 @@ public class HibernateReactiveTestEndpoint {
     @Path("/reactiveRemoveManagedEntity")
     @Produces(MediaType.APPLICATION_JSON)
     public Uni<String> reactiveRemoveManagedEntity() {
-        return mutinySession
-                .flatMap(mutinySession -> {
-                    return populateDB()
-                            .flatMap(junk -> mutinySession.find(GuineaPig.class, 5))
-                            .flatMap(aloi -> mutinySession.remove(aloi))
-                            .flatMap(junk -> mutinySession.flush())
-                            .flatMap(junk -> selectNameFromId(5))
-                            .map(result -> {
-                                if (result == null)
-                                    return "OK";
-                                else
-                                    return result;
-                            });
+        return populateDB()
+                .then(() -> mutinySession.find(GuineaPig.class, 5))
+                .chain(aloi -> mutinySession.remove(aloi))
+                .then(() -> mutinySession.flush())
+                .then(() -> selectNameFromId(5))
+                .map(result -> {
+                    if (result == null)
+                        return "OK";
+                    else
+                        return result;
                 });
     }
 
@@ -129,25 +116,23 @@ public class HibernateReactiveTestEndpoint {
     @Produces(MediaType.APPLICATION_JSON)
     public Uni<String> reactiveUpdate() {
         final String NEW_NAME = "Tina";
-        return mutinySession
-                .flatMap(mutinySession -> {
-                    return populateDB()
-                            .flatMap(junk -> mutinySession.find(GuineaPig.class, 5))
-                            .map(pig -> {
-                                if (NEW_NAME.equals(pig.getName()))
-                                    throw new AssertionError("Pig already had name " + NEW_NAME);
-                                pig.setName(NEW_NAME);
-                                return pig;
-                            })
-                            .flatMap(junk -> mutinySession.flush())
-                            .flatMap(junk -> selectNameFromId(5));
-                });
+        return populateDB()
+                .then(() -> mutinySession.find(GuineaPig.class, 5))
+                .invoke(pig -> {
+                    if (NEW_NAME.equals(pig.getName()))
+                        throw new AssertionError("Pig already had name " + NEW_NAME);
+                    pig.setName(NEW_NAME);
+                })
+                .then(() -> mutinySession.flush())
+                .then(() -> selectNameFromId(5));
     }
 
     private Uni<RowSet<Row>> populateDB() {
-        return pgPool.query("DELETE FROM Pig").execute()
-                .and(pgPool.query("DELETE FROM Cow").execute())
-                .flatMap(junk -> pgPool.preparedQuery("INSERT INTO Pig (id, name) VALUES (5, 'Aloi')").execute());
+        return Uni.combine().all().unis(
+                pgPool.query("DELETE FROM Pig").execute(),
+                pgPool.query("DELETE FROM Cow").execute())
+                .asTuple()
+                .then(() -> pgPool.preparedQuery("INSERT INTO Pig (id, name) VALUES (5, 'Aloi')").execute());
     }
 
     private Uni<String> selectNameFromId(Integer id) {

